R/eesapi_url.R

```diff
@@ -67,24 +82,84 @@ eesapi_url <- function(
     }
   }
 
+  if (environment == "dev") {
+    api_stub <- "https://dev.statistics.api.education.gov.uk/api/"
+  } else {
+    api_stub <- "https://statistics.api.education.gov.uk/api/"
+  }
+  api_stub_vers <- paste0(
+    api_stub,
+    "v", api_version, "/"
+  )
 
+  if (endpoint == "get-publications") {
+    url <- paste0(
+      api_stub_vers,
+      "publications?",
+      eesapi_url_pages(page_size = page_size, page = page)
+    )
+  } else if (endpoint == "get-data-catalogue") {
+    url <- paste0(
+      api_stub_vers,
+      "publications/",
+      publication_id,
+      "/data-sets?",
+      eesapi_url_pages(page_size = page_size, page = page)
+    )
+  } else {
+    url <- paste0(
+      api_stub_vers,
+      "data-sets/",
+      ifelse(
+        endpoint %in% c("get-summary", "get-meta", "get-data", "query-data"),
+        dataset_id,
+        ""
+      ),
+      ifelse(
+        endpoint %in% c("get-summary", "get-meta", "query-data"),
+        paste0("/", gsub("get-|-data", "", endpoint)),
+        ""
+      ),
+      ifelse(
+        !is.null(dataset_version),
+        paste0("?dataSetVersion=", dataset_version),
+        ""
+      )
+    )
+  }
+  if (verbose) {
+    cat("Generated the following query url:", fill = TRUE)
+    cat(url, fill = TRUE)
+  }
+  return(url)
+}
+
+#' EES API URL paging
+#'
```

Comment:
Description of where / why you might use this?

Reply:
Included a bit of context